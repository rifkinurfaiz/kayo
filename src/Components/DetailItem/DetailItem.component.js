import * as React from 'react';
import { View, Text } from 'react-native';

import styles from './DetailItem.styles';
import config from '../../Config';

const _dot = id => <View style={styles.dot(config.color[id])} />;

const _mapAccounts = accounts =>
  accounts.map(account => {
    return (
      <View style={styles.accounts}>
        {_dot(account.id)}
        <Text style={styles.accountName}>{account.name}</Text>
        <Text style={styles.amount(account.amount)}>{account.amount}</Text>
      </View>
    );
  });

export const DetailItem = props => (
  <View style={styles.container}>
    <View style={styles.title}>
      <Text style={styles.textTitle}>{props.month}</Text>
    </View>
    <View style={styles.accountsContainer}>{_mapAccounts(props.accounts)}</View>
    <Text>{props.totalAmount}</Text>
  </View>
);

export default DetailItem;
