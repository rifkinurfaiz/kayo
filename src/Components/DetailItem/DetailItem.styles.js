import { Dimensions } from 'react-native';
import memoize from 'fast-memoize';

import colors from '../../Assets/Colors';

const { width } = Dimensions.get('window');
const { primary } = colors;

export default {
  container: {
    borderRadius: 5,
    width: width / 2 - 20,
    backgroundColor: primary.white,
    marginBottom: 15
  },
  title: {
    padding: 7,
    alignItems: 'center',
    borderTopRightRadius: 5,
    borderTopLeftRadius: 5,
    backgroundColor: primary.blue
  },
  textTitle: {
    color: primary.white,
    fontWeight: 'bold'
  },
  accounts: {
    padding: 6,
    flexDirection: 'row',
    justifyContent: 'flex-start'
  },
  accountsContainer: {
    paddingHorizontal: 5
  },
  accountName: {
    width: 55,
    fontSize: 13
  },
  amount: memoize(amount => ({
    fontSize: amount.length > 15 ? 12 : 14
  })),
  dot: memoize(color => ({
    width: 10,
    height: 10,
    borderRadius: 5,
    backgroundColor: color,
    marginVertical: 5,
    marginRight: 5
  })),
  totalAmountContainer: {
    marginTop: 5,
    paddingVertical: 5,
    borderTopWidth: 1,
    borderTopColor: primary.grey,
    justifyContent: 'center',
    alignItems: 'center'
  },
  totalAmountText: {
    fontWeight: 'bold'
  }
};
