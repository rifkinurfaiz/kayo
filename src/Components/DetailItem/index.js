export { default } from './DetailItem.component';
