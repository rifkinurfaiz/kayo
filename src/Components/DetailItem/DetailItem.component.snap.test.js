import * as React from 'react';
import { shallow } from 'enzyme';

import DetailItem from './DetailItem.component';

describe('#DetailItemComponent', () => {
  it('render with font size of amount is 12', () => {
    const props = {
      month: 'January',
      assets: [
        {
          id: 'jenius',
          name: 'Jenius',
          amount: 'Rp1000.000'
        }
      ]
    };

    expect(shallow(<DetailItem {...props} />)).toMatchSnapshot();
  });

  it('render with font size of amount is 14', () => {
    const props = {
      month: 'January',
      assets: [
        {
          id: 'jenius',
          name: 'Jenius',
          amount: 'Rp1000.000.000.000'
        }
      ]
    };

    expect(shallow(<DetailItem {...props} />)).toMatchSnapshot();
  });
});
