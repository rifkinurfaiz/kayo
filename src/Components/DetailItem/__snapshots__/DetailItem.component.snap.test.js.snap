// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#DetailItemComponent render with font size of amount is 12 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "marginBottom": 15,
      "width": 355,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1e90ff",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "padding": 7,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontWeight": "bold",
        }
      }
    >
      January
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 5,
      }
    }
  >
    <View
      key="0"
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 6,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#01ABE9",
            "borderRadius": 5,
            "height": 10,
            "marginRight": 5,
            "marginVertical": 5,
            "width": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 13,
            "width": 55,
          }
        }
      >
        Jenius
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        Rp1000.000
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#e1e1e1",
        "borderTopWidth": 1,
        "justifyContent": "center",
        "marginTop": 5,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  </View>
</View>
`;

exports[`#DetailItemComponent render with font size of amount is 14 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "marginBottom": 15,
      "width": 355,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1e90ff",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "padding": 7,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontWeight": "bold",
        }
      }
    >
      January
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 5,
      }
    }
  >
    <View
      key="0"
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 6,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#01ABE9",
            "borderRadius": 5,
            "height": 10,
            "marginRight": 5,
            "marginVertical": 5,
            "width": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 13,
            "width": 55,
          }
        }
      >
        Jenius
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Rp1000.000.000.000
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#e1e1e1",
        "borderTopWidth": 1,
        "justifyContent": "center",
        "marginTop": 5,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  </View>
</View>
`;
